From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nathan Schulte <nmschulte@desmas.net>
Date: Sun, 15 Oct 2023 03:13:21 -0500
Subject: [PATCH] 5884: CParser: prevent crash when parsing

if the trimmed token length is zero, charAt() explodes

this is likely a bad macro, but CParser should not crash regardless
---
 .../src/main/java/ghidra/app/util/cparser/CPP/DefineTable.java | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/cparser/CPP/DefineTable.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/cparser/CPP/DefineTable.java
index 3030f9d7a3..70049f4b35 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/cparser/CPP/DefineTable.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/cparser/CPP/DefineTable.java
@@ -878,9 +878,8 @@ public class DefineTable {
 			pos = startPos;
 			int endParen = strValue.indexOf(')', pos + 1);
 			if (endParen != -1) {
-				String subStr = strValue.substring(pos + 1, endParen);
+				String subStr = strValue.substring(pos + 1, endParen).trim();
 				if (!subStr.isEmpty()) {
-					subStr = subStr.trim();
 					boolean isValid = true;
 					for (int subPos = 0; subPos < subStr.length(); subPos++) {
 						if (!Character.isJavaIdentifierStart(subStr.charAt(subPos))) {
-- 
2.45.1

